com.unity.render-pipelines.universal/Editor/2D/Converter/ParametricToFreeformLightUpgrader.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.IO;
+using System.Collections.Generic;
+using UnityEditor.Rendering.Universal.Converters;
+using UnityEngine;
+using UnityEngine.Rendering.Universal;
+
+
+namespace UnityEditor.Rendering.Universal
+{
+    internal sealed class ParametricToFreeformLightUpgrader : RenderPipelineConverter
+    {
+        public override string name => "Parametric to Freeform Light Upgrade";
+        public override string info => "This will upgrade all parametric lights to freeform lights.";
+        public override int priority => - 1000;
+        public override Type container => typeof(UpgradeURP2DAssetsContainer);
+
+        List<string> m_AssetsToConvert = new List<string>();
+
+        string m_Light2DId;
+
+        public static void UpgradeParametricLight(Light2D light)
+        {
+            if (light.lightType == (Light2D.LightType)Light2D.DeprecatedLightType.Parametric)
+            {
+                light.lightType = Light2D.LightType.Freeform;
+
+                float radius = light.shapeLightParametricRadius;
+                float angle = light.shapeLightParametricAngleOffset;
+                int sides = light.shapeLightParametricSides;
+
+                var angleOffset = Mathf.PI / 2.0f + Mathf.Deg2Rad * angle;
+                if (sides < 3)
+                {
+                    radius = 0.70710678118654752440084436210485f * radius;
```

Comment:
magic number here. Put it as a constant and name it to something that says what it is.

Reply:
Fixed